pymc3/distributions/dist_math.py

```diff
@@ -634,3 +470,224 @@ def clipped_beta_rvs(a, b, size=None, random_state=None, dtype="float64"):
+    F2 = (p + q + n - 2) * (p + n - 1) * (q - n) / ((p2n - 3) * (p2n - 2) ** 2 * (p2n - 1))
+
+    return F1 * F2
+
+
+def _betainc_b_n(f, p, q, n):
+    """
+    Offset (b_n) of the nth approximant of the continued fraction
+    representation of the regularized incomplete beta function
+    """
+    pf = p * f
+    p2n = p + 2 * n
+
+    N1 = 2 * (pf + 2 * q) * n * (n + p - 1) + p * q * (p - 2 - pf)
+    D1 = q * (p2n - 2) * p2n
+
+    return N1 / D1
+
+
+def _betainc_da_n_dp(f, p, q, n):
+    """
+    Derivative of a_n wrt p
+    """
+
+    if n == 1:
+        return -p * f * (q - 1) / (q * (p + 1) ** 2)
+
+    pp = p ** 2
+    ppp = pp * p
+    p2n = p + 2 * n
+
+    N1 = -(n - 1) * f ** 2 * pp * (q - n)
+    N2a = (-8 + 8 * p + 8 * q) * n ** 3
+    N2b = (16 * pp + (-44 + 20 * q) * p + 26 - 24 * q) * n ** 2
+    N2c = (10 * ppp + (14 * q - 46) * pp + (-40 * q + 66) * p - 28 + 24 * q) * n
+    N2d = 2 * pp ** 2 + (-13 + 3 * q) * ppp + (-14 * q + 30) * pp
+    N2e = (-29 + 19 * q) * p + 10 - 8 * q
+
+    D1 = q ** 2 * (p2n - 3) ** 2
+    D2 = (p2n - 2) ** 3 * (p2n - 1) ** 2
+
+    return (N1 / D1) * (N2a + N2b + N2c + N2d + N2e) / D2
+
+
+def _betainc_da_n_dq(f, p, q, n):
+    """
+    Derivative of a_n wrt q
+    """
+    if n == 1:
+        return p * f / (q * (p + 1))
+
+    p2n = p + 2 * n
+    F1 = (p ** 2 * f ** 2 / (q ** 2)) * (n - 1) * (p + n - 1) * (2 * q + p - 2)
+    D1 = (p2n - 3) * (p2n - 2) ** 2 * (p2n - 1)
+
+    return F1 / D1
+
+
+def _betainc_db_n_dp(f, p, q, n):
+    """
+    Derivative of b_n wrt p
+    """
+    p2n = p + 2 * n
+    pp = p ** 2
+    q4 = 4 * q
+    p4 = 4 * p
+
+    F1 = (p * f / q) * ((-p4 - q4 + 4) * n ** 2 + (p4 - 4 + q4 - 2 * pp) * n + pp * q)
+    D1 = (p2n - 2) ** 2 * p2n ** 2
+
+    return F1 / D1
+
+
+def _betainc_db_n_dq(f, p, q, n):
+    """
+    Derivative of b_n wrt to q
+    """
+    p2n = p + 2 * n
+    return -(p ** 2 * f) / (q * (p2n - 2) * p2n)
+
+
+def _betainc_derivative(x, p, q, wrtp=True):
+    """
+    Compute the derivative of regularized incomplete beta function wrt to p (alpha) or q (beta)
+
+    Reference: Boik, R. J., & Robison-Cox, J. F. (1998). Derivatives of the incomplete beta function.
+    Journal of Statistical Software, 3(1), 1-20.
+    """
+
+    # Input validation
+    if not (0 <= x <= 1) or p < 0 or q < 0:
+        return np.nan
+
+    if x > (p / (p + q)):
+        return -_betainc_derivative(1 - x, q, p, not wrtp)
+
+    min_iters = 3
+    max_iters = 200
+    err_threshold = 1e-12
+
+    derivative_old = 0
+
+    Am2, Am1 = 1, 1
+    Bm2, Bm1 = 0, 1
+    dAm2, dAm1 = 0, 0
+    dBm2, dBm1 = 0, 0
+
+    f = (q * x) / (p * (1 - x))
+    K = np.exp(p * np.log(x) + (q - 1) * np.log1p(-x) - np.log(p) - scipy.special.betaln(p, q))
+    if wrtp:
+        dK = np.log(x) - 1 / p + scipy.special.digamma(p + q) - scipy.special.digamma(p)
+    else:
+        dK = np.log1p(-x) + scipy.special.digamma(p + q) - scipy.special.digamma(q)
+
+    for n in range(1, max_iters + 1):
```

Comment:
Is the plain python for loop going to be slow here, or is the function is meant to run in python/numpy/scipy

Reply:
Main Python loop for now. It's not really possible to vectorize as it is an iterative loop. It's still about ~4-5 orders of magnitude faster than the old derivatives obtained by autodiff from the scan implementation of the `incomplete_beta`.

It's also very simple code to port to `C` or to `Numba` but I didn't want to do it just yet.

I have some speed comparisons here https://www.github.com/ricardoV94/derivatives_betainc/tree/master/comparison_aesara.ipynb